pennylane/spin/lattice.py

```diff
@@ -54,6 +60,7 @@ class Lattice:
           if ``positions`` doesn't have a dimension of 2.
           if ``vectors`` doesn't have a dimension of 2 or the length of vectors is not equal to the number of vectors.
           if ``boundary_condition`` is not a bool or a list of bools with length equal to the number of vectors
+          if ``custom_nodes`` contains nodes with negative indices or indices greater than number of sites
```

Comment:
To confirm, I can have multiple instances of the same node index with different operator right?

Reply:
Yes. No issue with that.

Reply:
Although perhaps we should simplify the hamiltonians at the end of the custom spin_hamiltonian function.